### ISSUE TITLE
[BUG] action works on Safari, but fails in WebKit

### ISSUE THREAD

**Issue by guettli** [microsoft/playwright#5894]
**Context:**
- Playwright Version: 1.9.2
- Operating System: Ubuntu 20.04
- Node.js version: I use Python 3.8.5
- Browser: WebKit
- Extra: URL to reproduce: https://thomas-guettler.de/htmx-swap-err-webkit/page.html

**Code Snippet**

```
const { webkit } = require('playwright');

(async () => {
  const browser = await webkit.launch({
    headless: false
  });
  const context = await browser.newContext();

  // Open new page
  const page = await context.newPage();

  // Go to https://thomas-guettler.de/htmx-swap-err-webkit/page.html
  await page.goto('https://thomas-guettler.de/htmx-swap-err-webkit/page.html');

  // Click textarea[name="comment"]
  await page.click('textarea[name="comment"]');

  // Fill textarea[name="comment"]
  await page.fill('textarea[name="comment"]', 'foobar');

  // Click text=Save
  await page.click('text=Save');

  // ---------------------
  await context.close();
  await browser.close();
})();
```

**Describe the bug**

If I press "Save" nothing happens. But finally you should see "Great, delicious food is on its way to you"

This works with chromium. And it works with Safari on MacOS.

Please let me know if you need more info.

**Comment by yury-s** [comment]
Running ` ~/.cache/ms-playwright/webkit-1446/pw_run.sh https://thomas-guettler.de/htmx-swap-err-webkit/page.html` on linux results in a segfault for me:

```
(gdb) bt
#0  0x00007f52e7583a5b in WTF::Detail::CallableWrapper<WebKit::IconDatabase::loadIconForPageURL(WTF::String const&, WebKit::IconDatabase::AllowDatabaseWrite, WTF::CompletionHandler<void (WTF::RefPtr<_cairo_surface, WTF::RawPtrTraits<_cairo_surface>, WTF::DefaultRefDerefTraits<_cairo_surface> >&&)>&&)::$_3::operator()()::{lambda()#1}, void>::call() () at /home/yurys/WebKit/WebKitBuild/GTK/Release/lib/libwebkit2gtk-4.0.so.37
#1  0x00007f52e5999e80 in WTF::RunLoop::performWork() () at /home/yurys/WebKit/WebKitBuild/GTK/Release/lib/libjavascriptcoregtk-4.0.so.18
#2  0x00007f52e5a00be6 in WTF::RunLoop::RunLoop()::$_1::__invoke(void*) () at /home/yurys/WebKit/WebKitBuild/GTK/Release/lib/libjavascriptcoregtk-4.0.so.18
#3  0x00007f52e5a000fa in WTF::RunLoop::$_0::__invoke(_GSource*, int (*)(void*), void*) () at /home/yurys/WebKit/WebKitBuild/GTK/Release/lib/libjavascriptcoregtk-4.0.so.18
#4  0x00007f52eabb804e in g_main_context_dispatch () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#5  0x00007f52eabb8400 in  () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#6  0x00007f52eabb84a3 in g_main_context_iteration () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#7  0x00007f52e1329fe5 in g_application_run () at /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0
#8  0x00000000004177c7 in main ()
```
it is likely related. I'll try to fix that and see if the problem persists. Also the code works just fine with Playwright on macOS (with the bundled webkit version), @guettli is that what you meant by saying that it works with Safari?

**Comment by yury-s** [comment]
With a quick fix for the icon  loader I now see the following error printed to js console:

![image](https://user-images.githubusercontent.com/9798949/112044350-f7e55a80-8b06-11eb-9b0c-df99afe838c3.png)

**Comment by pavelfeldman** [comment]
Nothing crashes for me, but I can confirm that running this on macOS works fine, while running it on Ububtu 20 leads to the error above in console.

**Comment by pavelfeldman** [comment]
I think you just found a browser-specific bug in your code (htmx to be exact) with Playwright:
Tl/dr, tab to Submit and press Space in Safari. It'll click but won't work. You will see error in console.

Details:
Upon successful form submit, it uses document.activeElement and calls selectionStart on it. In Chromium, it is going to be `<input>`, because you just pressed Submit input. Same goes for WebKit on Ubuntu. Interestingly, on Safari on a Mac, it is `<body>` if you tap, not sure why, but probably input was swapped for another element and body became active element.

Now calling `input.selectionStart` on submit input in Chromium is Ok, but on WebKit it throws. So the line 2084 throws and the lines 2087-2088 that actually render the new screen never happen.

<img width="432" alt="Screen Shot 2021-03-29 at 4 25 29 PM" src="https://user-images.githubusercontent.com/883973/112911998-66985a00-90ab-11eb-8674-b472fa5fcde0.png">

I don't see any error handling inside the catch for this try, so the reason it works in Safari is pure luck.

**Comment by guettli** [comment]
@pavelfeldman thank you very much for your reply. I will talk to the htmx creators:

https://github.com/bigskysoftware/htmx/issues/438

**Comment by pavelfeldman** [comment]
Closing as a non-actionable bug, please feel free to file a new one and link this bug if there are remaining questions.
